### PR TITLE
Unit tests for the Rook operator

### PR DESCRIPTION
**Pull request by travisn**

### CODE REVIEW

File: pkg/operator/api/api.go [hunk 1]
```diff
@@ -125,12 +126,13 @@ func (c *Cluster) apiContainer(cluster *mon.ClusterInfo) v1.Container {
...
 
-func (c *Cluster) startService(clientset *kubernetes.Clientset, clusterInfo *mon.ClusterInfo) error {
+func (c *Cluster) startService(clientset kubernetes.Interface, clusterInfo *mon.ClusterInfo) error {
 	labels := getLabels(clusterInfo.Name)
 	s := &v1.Service{
 		ObjectMeta: v1.ObjectMeta{
-			Name:   deploymentName,
-			Labels: labels,
+			Name:      deploymentName,
+			Namespace: c.Namespace,
```
Comment (jbw976) [inline pkg/operator/api/api.go]: did namespace get added here because it was a bug exposed by the new tests?

File: pkg/operator/api/api_test.go [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+	info := testop.CreateClusterInfo(1)
+	c := New("ns", "myversion")
+
+	// start a basic cluster
+	err := c.Start(clientset, info)
+	assert.Nil(t, err)
+
+	validateStart(t, c, clientset)
+
+	// starting again should be a no-op, but still results in an error
```
Comment (jbw976) [inline pkg/operator/api/api_test.go]: comment says it results in an error but then we assert that err is nil.  Those are conflicting right?

i think this comment occurs in a couple other locations

File: pkg/operator/mon/mon_test.go [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+func validateStart(t *testing.T, c *Cluster, clientset *fake.Clientset) {
+	s, err := clientset.CoreV1().Secrets(c.Namespace).Get("rook-admin")
+	assert.Nil(t, err)
+	assert.Equal(t, 1, len(s.StringData))
+
+	s, err = clientset.CoreV1().Secrets(c.Namespace).Get("mon")
+	assert.Nil(t, err)
+	assert.Equal(t, 4, len(s.StringData))
+
+	// there is only one pod created. the other two won't be created since the first one doesn't start
```
Comment (jbw976) [inline pkg/operator/mon/mon_test.go]: is there more useful testing that can be done around getting the rods running/started too?

File: pkg/operator/api/api.go (hunk 1 above)
Comment (travisn) [inline pkg/operator/api/api.go]: the mocked framework was more strict about setting the namespace on pods and services. I haven't seen a difference in behavior in the real world though.

File: pkg/operator/api/api_test.go (hunk 2 above)
Comment (travisn) [inline pkg/operator/api/api_test.go]: will clean up the comments thanks

File: pkg/operator/mon/mon_test.go (hunk 3 above)
Comment (travisn) [inline pkg/operator/mon/mon_test.go]: i didn't see a way with the mocked api to get the pods in running state. i'd really like to see that too though.

**Pull request merged**